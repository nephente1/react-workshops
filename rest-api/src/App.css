.App {
    display: flex;
    flex-direction: column;
    width: 100vw;
    justify-content: flex-start;;
    background-color: #282c34;
    min-height: 100vh;
}

.main-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    text-align: center;
    font-size: 14px;
    color: white;
}

.App-link {
    color: #61dafb;
}

.userBox {
    border: 1px solid silver;
    padding: 15px;
    margin: 15px;
    text-align: center;
    float: left;
    color: white;
    width: 300px;
    height: 250px;
}

.userContainer {
    display: flex;
    justify-content: center;
    text-align: center;
    font-size: 14px;
    color: white;
    flex-direction: column;
    align-items: center;
}

h1 {
    color: white;
    text-align: center;
}
