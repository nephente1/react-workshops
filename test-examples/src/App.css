.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  color: #fff;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.App-link {
  color: #61dafb;
}


