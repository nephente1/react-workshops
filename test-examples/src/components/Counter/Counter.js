import React from 'react';

export const Counter = (props) => {
    const startCounter = props.start !== undefined ? props.start : 0;
    const [value, setValue] = React.useState(props.start);
    const [counter, setCounter] = React.useState(startCounter);

    const handleAdd = () => {
        setCounter(Number(counter)+1);
    }

    const handleSubtract = () => {
        setCounter(counter-1);
    }

    const handleReset = () => {
        setCounter(props.start);
        setValue(props.start);
    }

    const handleChange = (event) => {
        const newValue = event.target.value;
        setValue(newValue);
    }

    const changeStartValue = () => {
        setCounter(value)
    }

    return(
        <div>
            <h2>Wartość początkowa licznika: <span>{value}</span></h2>
            <h3>Licznik: <span>{counter}</span></h3>
            <input onChange={handleChange} type="number" value={value} placeholder='type start number'/>
            <button data-test='changeButton' onClick={changeStartValue}>zmień</button>
            <button data-test='addButton' onClick={handleAdd}>+</button>
            <button data-test='decrementButton' onClick={handleSubtract}>-</button>
            <button data-test='resetButton' onClick={handleReset}>reset</button>
        </div>
    )
}
