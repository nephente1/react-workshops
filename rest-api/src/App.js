import React from 'react';
import './App.css';
import moment from "moment";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { useHistory } from 'react-router-dom';

import ReactPlaceholder from 'react-placeholder';
import "react-placeholder/lib/reactPlaceholder.css";
import {TextBlock, RoundShape} from 'react-placeholder/lib/placeholders';
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';


const Context = React.createContext();

const App = () => {
	const [context, setContext] = React.useState(); //for context
	const [data, setData] = React.useState();

	React.useEffect( () => {
		const loadData = async() => {
			const response = await fetch(`https://randomuser.me/api/?results=10`);
			const respJson = await response.json();
			setData(respJson.results);
		}
		setTimeout(() => {
			loadData()
		}, 300);

	}, []);

	return (
		<div className="App">
			<h1>Random Users App</h1>

			<Context.Provider value={[context, setContext]}> {/* passing functions to create context */}
				<Router>
					<Switch>
						<Route exact path="/" render={() => <UserList data={data} />}/>
						<Route path="/:id" children={<UserDetails />}/>
					</Switch>
				</Router>
			</Context.Provider>

		</div>
	);
}
export default App;


const UserList = (props) => {
	const awesomePlaceholder = (
		<div className='my-awesome-placeholder' style={{width: '80vw'}}>
		    <RoundShape color='silver' style={{width: 120, height: 120, margin: '0 0 20px 0'}}/>
    		<TextBlock rows={7} color='silver'/>
		</div>
	);
	return(
		<div className="main-list">
			<ReactPlaceholder showLoadingAnimation ready={props.data !== undefined} customPlaceholder={awesomePlaceholder}>
			{props.data?.map(el =>
				<User
					key={el.login.uuid}
					name={el.name.first}
					surname={el.name.last}
					adress={el.location.city + ' - ' + el.location.country}
					email={el.email}
					registration={el.registered.date}
					image={el.picture}
					uuid={el.login.uuid}
					location={el.location}
				/>)
			}
			</ReactPlaceholder>
		</div>
	)
}


const User = (props) => {
	const [context, setContext] = React.useContext(Context);
	const { name, surname, adress, email, registration, image, uuid } = props;
	const dateFormated = moment(registration).format('YYYY-MM-DD HH:ss');
	return(
		<Link to={{ pathname: `${uuid}` }}>
			<div onClick={() => setContext(props)} className='userBox'> {/* creating context object with data for clicked userDetails */}
				<p>Name: {name}</p>
				<p>Surname: {surname}</p>
				<p>Adress: {adress}</p>
				<p>Email: {email}</p>
				<p>Registration date: {dateFormated}</p>
				<div><img alt={name} src={image.thumbnail} /></div>
			</div>
		</Link>
	)
}


const UserDetails = () => {
	let history = useHistory();
	const handleBackClick = () => {
		history.push('/');
	}

	const [context] = React.useContext(Context);
	const position = [context.location.coordinates.latitude, context.location.coordinates.longitude];
	return(
		<div className='userContainer'>
			<h1>{context.name} {context.surname}</h1>
			<p><img alt={context.name} src={context.image.large}/></p>
			<h3>City: {context.location.city}</h3>

			<MapContainer style={{width: '90%', height: '300px'}} center={position} zoom={2} scrollWheelZoom={true}>
				<TileLayer
					attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
					url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
					/>
				<Marker position={position}>
					<Popup>
					Hello from <br /> {context.location.city}!
					</Popup>
				</Marker>
			</MapContainer>
			<button onClick={handleBackClick}>back</button>
		</div>
	)
}

