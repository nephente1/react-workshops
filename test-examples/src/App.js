import './App.css';
import { AddComponent } from './components/AddComponent/AddComponent';
import { Counter } from './components/Counter/Counter';

function App() {
  return (
    <div className="App">
      <AddComponent />
      <Counter start={0} />
    </div>
  );
}

export default App;

//npm test -- -u => przerenderowanie testu
