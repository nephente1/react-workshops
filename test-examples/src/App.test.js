
import { render, screen } from '@testing-library/react';
import { shallow, mount, enzyme } from 'enzyme';
import App from './App';
import { Counter } from './components/Counter/Counter';

describe('<App><Counter /></App>', () => {
  const wrapper = shallow((<App />));
  it('renders Counter in App', () => {
    expect(wrapper.find(Counter)).toHaveLength(1);
  })

  it('Snapshot app.js', () => {
      expect(wrapper).toMatchSnapshot();
  });

});
