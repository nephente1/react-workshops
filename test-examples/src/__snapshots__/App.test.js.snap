// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App><Counter /></App> Snapshot app.js 1`] = `ShallowWrapper {}`;
